### PR TITLE
RETRO_ENVIRONMENT_SET_ACTIVE_PORT_COUNT

### PR DESCRIPTION
**Pull request by hiddenasbestos**
#define RETRO_ENVIRONMENT_SET_ACTIVE_PORT_COUNT (50 | RETRO_ENVIRONMENT_EXPERIMENTAL)

uint32_t *

Provides a count to the front-end stating how many controller ports are 
actively being monitored by the core. This may be fewer than the highest
port value in the RETRO_ENVIRONMENT_SET_INPUT_DESCRIPTORS table, or the 
number of entries in the RETRO_ENVIRONMENT_SET_CONTROLLER_INFO table.
The value can change at any time - usually in response to a change in
input device configuration. For example connecting (or enabling via core
option) a multitap might raise the port count from 2 to 5. Alternatively
specific games might provide additional controller hardware - e.g. 
Micro Machines 2 on the Mega Drive has additional controller ports built
into the cartridge.
If this value has not been set, use the existing method of counting the
RETRO_ENVIRONMENT_SET_CONTROLLER_INFO table size.

### CODE REVIEW

**Comment by garbear** [comment]
I have a lot to say on this issue. I'll do some work when I have time and return in a few days.

**Comment by hiddenasbestos** [comment]
Just spotted that Reicast uses id 49 for RETRO_ENVIRONMENT_GET_FASTFORWARDING, so don't merge this yet, please.

Would still like to discuss this feature though. Having a lightweight way for the core to update the front-end on what ports are available seems useful.

**Comment by hiddenasbestos** [comment]
Hi @garbear - what are your comments?

**Comment by hiddenasbestos** [comment]
I've fixed the conflict with Reicast and pushed a new version of libretro,h that uses 50 for the environment code for this new feature.

File: include/libretro.h [hunk 1]
```diff
@@ -1069,7 +1069,12 @@ enum retro_mod
                                            /* struct retro_midi_interface ** --
                                             * Returns a MIDI interface that can be used for raw data I/O.
                                             */
-#define RETRO_ENVIRONMENT_SET_ACTIVE_PORT_COUNT (49 | RETRO_ENVIRONMENT_EXPERIMENTAL)
+#define RETRO_ENVIRONMENT_GET_FASTFORWARDING (49 | RETRO_ENVIRONMENT_EXPERIMENTAL)
+                                            /* bool * --
+                                            * Boolean value that indicates whether or not the frontend is in
+                                            * fastforwarding mode.
+                                            */
+#define RETRO_ENVIRONMENT_SET_ACTIVE_PORT_COUNT (50 | RETRO_ENVIRONMENT_EXPERIMENTAL)
```
Comment (garbear) [inline include/libretro.h]: Just a comment on PR organization, but if you invert the order of these two commits you won't have a commit deleting an unrelated line.

File: include/libretro.h [hunk 2]
```diff
@@ -1069,7 +1069,12 @@ enum retro_mod
                                            /* struct retro_midi_interface ** --
                                             * Returns a MIDI interface that can be used for raw data I/O.
                                             */
-#define RETRO_ENVIRONMENT_SET_ACTIVE_PORT_COUNT (49 | RETRO_ENVIRONMENT_EXPERIMENTAL)
```
Comment (garbear) [inline include/libretro.h]: Just a comment on PR organization, but if you invert the order of these two commits you won't have a commit deleting an unrelated line.

**Comment by hiddenasbestos** [comment]
Decided to abandon this idea!
